src/SpectralDataDialog.cpp

```diff
@@ -0,0 +1,251 @@
+               .Name(XO("Custom brush size"))
+               .AddSlider( {}, 5, 10, 1);
+      }
+      S.EndStatic();
+   }
+   mApplyBtn = S.Id(ID_ON_APPLY)
+         .AddButton(XXO("Apply effect to selection."));
+   S.EndVerticalLay();
+   // ----------------------- End of main section --------------
+
+   Layout();
+   Fit();
+   SetMinSize(GetSize());
+}
+
+void SpectralDataDialog::OnAudioIO(wxCommandEvent& evt)
+{
+   evt.Skip();
+
+   if (evt.GetInt() != 0)
+      mAudioIOBusy = true;
+   else
+      mAudioIOBusy = false;
+}
+
+void SpectralDataDialog::UpdateDisplay(wxEvent& e)
+{
+   e.Skip();
+   if(IsShown())
+      DoUpdate();
+}
+
+bool SpectralDataDialog::Show( bool show )
+{
+   if ( show && !IsShown())
+      DoUpdate();
+   return wxDialogWrapper::Show( show );
+}
+
+void SpectralDataDialog::DoUpdate()
+{
+}
+
+void SpectralDataDialog::OnCloseWindow(wxCloseEvent & WXUNUSED(event))
+{
+   this->Show(false);
+}
+
+void SpectralDataDialog::OnShow(wxShowEvent & event)
+{
+   if (event.IsShown())
+   {
+//      mApplyBtn->SetFocus();
+   }
+}
+
+// PrefsListener implementation
+void SpectralDataDialog::UpdatePrefs()
+{
+   bool shown = IsShown();
+   if (shown) {
+      Show(false);
+   }
+
+   SetSizer(nullptr);
+   DestroyChildren();
+
+   SetTitle(Title);
+   ShuttleGui S(this, eIsCreating);
+   Populate(S);
+
+   if (shown) {
+      Show(true);
+   }
+}
+
+void SpectralDataDialog::OnApply(wxCommandEvent &event) {
+   auto &tracks = TrackList::Get(*mProject);
+   auto &trackPanel = TrackPanel::Get(*mProject);
+
+   int applyCount = SpectralDataManager::ProcessTracks(tracks);
+   if (applyCount) {
+      trackPanel.Refresh(false);
+      AudacityMessageBox(XO("Effect applied to %d selection(s).").Format(applyCount));
+      ProjectHistory::Get(*mProject).PushState(
+            XO("Applied effect to selection"),
+            XO("Applied effect to selection"));
+      ProjectHistory::Get(*mProject).ModifyState(true);
+   }
+}
+
+static const AudacityProject::AttachedObjects::RegisteredFactory sSpectralWorkerKey{
+   []( AudacityProject &project ){
+      return std::make_shared< SpectralDataDialogWorker >( project );
+   }
+};
+
+AudacityProject::AttachedWindows::RegisteredFactory sSpectralDataDialogKey{
+      []( AudacityProject &parent ) -> wxWeakRef< wxWindow > {
+         return safenew SpectralDataDialog( &parent );
+      }
+};
+
+// Current workflow for spectral editing dialog:
+// 1. ProjectSettings change event listened by Worker::OnToolChanged()
+// 2. In Worker::OnToolChanged(), get Dialog from AttachedWindows and invoke Show()
+// 3. Dialog::OnApply() listens to the apply button in Dialog, which calculates and applies the effect
+SpectralDataDialogWorker::SpectralDataDialogWorker(AudacityProject &project)
+{
+   project.Bind(EVT_PROJECT_SETTINGS_CHANGE, &SpectralDataDialogWorker::OnToolChanged, this);
+   mProject = &project;
+}
+
+void SpectralDataDialogWorker::OnToolChanged(wxCommandEvent &evt)
+{
```

Comment:
This function fails to compile with EXPERIMENTAL_BRUSH_TOOL off.  Maybe you can simply wrap this whole file in an #ifdef to turn it off.  Likewise for the header.